src/gui/editorwidgets/qgsrangewidgetwrapper.cpp

```diff
@@ -132,65 +130,26 @@ void QgsRangeWidgetWrapper::initWidget( QWidget* editor )
       mIntSpinBox->setValue( minval );
       mIntSpinBox->setSpecialValueText( QSettings().value( "qgis/nullValue", "NULL" ).toString() );
     }
-    if ( min.isValid() )
-      mIntSpinBox->setMinimum( min.toInt() );
-    if ( max.isValid() )
-      mIntSpinBox->setMaximum( max.toInt() );
-    if ( step.isValid() )
-      mIntSpinBox->setSingleStep( step.toInt() );
+    mIntSpinBox->setMinimum( min.isValid() ? min.toInt() : std::numeric_limits<int>::min() );
+    mIntSpinBox->setMaximum( max.isValid() ? max.toInt() : std::numeric_limits<int>::max() );
+    mIntSpinBox->setSingleStep( step.isValid() ? step.toInt() : 1 );
     if ( config( "Suffix" ).isValid() )
       mIntSpinBox->setSuffix( config( "Suffix" ).toString() );
     connect( mIntSpinBox, SIGNAL( valueChanged( int ) ), this, SLOT( valueChanged( int ) ) );
   }
 
 
-  if ( mQgsDial || mQgsSlider )
+  if ( mQgsDial || mQgsSlider || mDial || mSlider )
   {
     field().convertCompatible( min );
     field().convertCompatible( max );
     field().convertCompatible( step );
-
-    if ( mQgsSlider )
```

Comment:
Oops, missed that. Weird stuff. OK, switching to using templates to work around the copy&paste...